Cargo.toml

```diff
@@ -10,6 +10,7 @@ description = "Starknet Rust types related to computation and execution."
 [features]
 default = ["std"]
 std = ["serde_json/std", "hex/std"]
+parity-scale-codec = ["dep:parity-scale-codec"]
```

Comment:
The way of using this crate I have seen the most widely used is to  import it as follows:
`scale-codec = ["dep:parity-scale-codec"]`

This way you can later use the less verbose:
#[cfg_attr(feature = "scale-codec", derive(Encode, Decode))]

Reply:
Also, you should specify that, when both the feature `std` and `scale-codec` are enabled, we should use the `parity-scale-codec` crate with it's `std` feature.
The way to do this is:
`std = ["serde_json/std", "hex/std", "parity-scale-codec?/std"]`

You refer here to the optional crate with its full name. And the `?` mean that it's std part is only imported if it is already present in the binary.

See: https://doc.rust-lang.org/cargo/reference/features.html#dependency-features for more about the `?` in this context